### ISSUE TITLE
Any logged in user can view any report.

### ISSUE THREAD

**Issue by danmarsden** [drillbit-dev/moodle-plagiarism_drillbit#11]
this file:
https://github.com/drillbit-dev/moodle-plagiarism_drillbit/blob/main/report_file.php

Appears to allow any user to just pass an id number and view a report - there are no checks to see if the user is enrolled in the course, is a teacher so can view all reports from that course, or is the student trying to access their own report.

**Comment by Kavimukil** [comment]
@danmarsden Now we have made changes to check whomever is viewing the report whether they are enrolled/not and teacher who created the assignment will be able to view. 

By default students will also be able to view their own reports.

**Comment by danmarsden** [comment]
Still looks wrong... you need to check can this student, view this paperid.

Eg - it still allows a student to pass in the paperid of another student in their course - essentially allowing them to view another students assignment.

your new "has_access_to_view_report" is also written in a way that doesn't really follow good practice - for starters, it embeds a function at the bottom of a standard file (functions should be in classes or lib.php/locallib.php style files) - also as mentioned before - functions must use the frankenstyle naming conventions etc...

Also - you have hard-coded role names in there "student/editingteacher" - this is really quite nasty and is a blocker for approval in the plugins db - many sites will not use those roles, or will have changed the default names etc - you need to be checking capabilities - eg something that checks if this user has the moodle capability to view all plagiarism reports in this course, or the logged in user matches the userid of this paperid.

**Comment by Kavimukil** [comment]
@danmarsden  Thank you for all the inputs and pointing us in the right way.

We were able to take the capabilities reference from urkund plugin to ensure we do not leave any security related loop holes.

As per your suggestions, roles are not hard-coded and We have brought capabilities to check if the logged in user is able to view report as per capabilities defined. 

Managers/Admin would be able to decide if reports needs to be visible for users (Below Screenshot ). User submitting the assignment will only be able to view his/her report as per assigned report Id. 

![image](https://user-images.githubusercontent.com/10370648/115557514-64eb3c00-a2cf-11eb-9dc4-9de345165376.png)

We have also included various test scenarios to check report access as per capabilities, We thank you again for guiding us on this.

Kindly request you to take a look at it.
